File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cs
public class Solution
{
    public int[] Intersection(int[] firstInput, int[] secondInput)
    {
        HashSet<int> presentInFirstInput = new HashSet<int>();
        foreach (var num in firstInput)
        {
            presentInFirstInput.Add(num);
        }

        List<int> intersection = new List<int>();
        foreach (var num in secondInput)
        {
            if (presentInFirstInput.Remove(num))
            {
                intersection.Add(num);
            }
        }

        return intersection.ToArray();
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.js
var intersection = function(nums1, nums2) {
    const num1Map = {}
    for(const num of nums1){
       num1Map[num] = true
    }
    const num2Map = {}

    for(const num of nums2){
        if(num in num1Map)num2Map[num] = true
    }
    return Object.keys(num2Map)
};